test/common/api_utils.py

```diff
@@ -155,29 +159,30 @@ def api_validate_node_pollers(client, node_id_list, all_pollers=False):
     return: True  - all nodes have active poller
             False - all nodes do not have active pollers
     """
-    pollers_list = []
-    good_pollers = True
-    # TODO: this will change in the near future
-    verbosity = int(os.getenv("VERBOSITY", "0"))
+    pollers_list = list()
 
     if not isinstance(node_id_list, (tuple, list, set)):
-        print '**** Provided node_list is not a [list]'
+        logs.error('Provided node_list is not a [list]')
+
         # return no good pollers
         return False
 
-    for node_ld in node_id_list:
-        Api().nodes_get_pollers_by_id(node_ld)
-        node_poller_rsp = client.last_response
-        node_poller_list = loads(node_poller_rsp.data)
-        if node_poller_rsp.status != 200:
-            print '**** Unable to retrieve node: {} pollers via API. status ({})\n'.format(node.get('id'),
-                                                                                           node_poller_rsp.status)
+    for node_id in node_id_list:
+        try:
+            Api().nodes_get_pollers_by_id(node_id)
+        except (TypeError, ValueError) as e:
+            assert (e.message)
+        except ApiException as e:
+            logs.debug('Unable to retrieve node: %s pollers via API. status (%s)', node_id, e.status)
```

Comment:
We should not be making asserts in a library, report the error back to the calling test script.